packages/network-controller/src/NetworkController.ts

```diff
@@ -728,13 +714,14 @@ export class NetworkController extends BaseControllerV2<
   }
 
   /**
-   * Fetches the network ID for the network, ensuring that it is a hex string.
-   *
+   * @deprecated Use of this method is discouraged as network ID is almost always
+   * only the concern of a node when establishing peer connections with other nodes.
+   * Chain ID is what should be used instead in the context of transactions and a wallet.
    * @returns A promise that either resolves to the network ID, or rejects with
    * an error.
    * @throws If the network ID of the network is not a valid hex string.
    */
-  async #getNetworkId(): Promise<NetworkId> {
+  async deprecatedGetNetworkId(): Promise<string> {
```

Comment:
reverted `getNetworkId` removal and made it a public method instead. Did this because extension's MetaMaskController still needs networkId for populating `networkVersion` in the `metamask_chainChanged` event

Reply:
See https://github.com/MetaMask/metamask-extension/pull/20652#issuecomment-1706980005

Reply:
removed it again. Team has decided to make this call from MetaMask controller